bcl2fastq/index_summary.py

```diff
@@ -98,99 +96,27 @@
-
-        Returns:
-            The string value (without '/') of the user input for the drop-down to use
-            Error pop-up displayed depending on user input.
-    """
-    #   In a pathname, it automatically adds '/' to the beginning of the input even if pathname blank
-    #   While page loads, pathname is set to 'None'. Once page is loaded pathname is set to user input.
-    if value == "/" or value is None:
-        return all_runs[0], False
-    elif value[1:] not in all_runs:
-        return all_runs[0], True
-    else:
-        return value[1:], False
-
-
-@app.callback(
-    dep.Output('sample_run_hidden', 'children'),
-    [dep.Input('run_select', 'value')]
-
-)
-def update_sample_run_hidden(run_alias):
-    """ When input (run_select) is changed, function is rerun to get json data for selected run
-        Parameters:
-            run_alias(str): alias of the run
-
-        Returns:
-             json object with columns, indices, and data, from run variable
-             updates json string "sample_run_hidden"
-"""
-    run = index[index['Run'] == run_alias]
-    run = run[run['ReadNumber'] == 1]
-    run = run[~run['SampleID'].isna()]
-    run = run.drop_duplicates(['SampleID', 'LaneNumber'])
-
-    return run.to_json(date_format='iso', orient='split')
-
-
-@app.callback(
-    dep.Output('pruned_unknown_hidden', 'children'),
-    [dep.Input('run_select', 'value')]
-)
-def update_pruned_unknown_hidden(run_alias):
-    """ When input (run_select) is changed, function prune_unknown_index_from_run is run within
-        update_pruned_unknown_hidden to get json data for selected run
-        Parameters:
-            run_alias(str): alias of the run
-
-        Returns:
-             json object with columns, indices, and data, from pruned variable
-             updates json string "pruned_unknown_hidden"
-    """
-    pruned = gsiqcetl.bcl2fastq.utility.prune_unknown_index_from_run(
-        run_alias, index, unknown
-    )
-    return pruned.to_json(date_format='iso', orient='split')
-
-
-@app.callback(
-    dep.Output('known_index_bar', 'figure'),
-    [dep.Input('sample_run_hidden', 'children')
-     ]
-)
-def update_known_index_bar(run_json):
-    """ When input (sample_run_hidden) is changed, function is rerun to get update bar graph of sample indices
+def update_known_index_bar(run_alias):
+    """ Function to update known index bar according to user selected run
            Parameters:
-               run_json(str): json format of run data
-
+               run_alias: name of run
            Returns:
               data and values for the layout of stacked bar graph of sample indices
               updates bar graph "known_index_bar"
        """
-    run = pandas.read_json(run_json, orient='split')
+    run = index[index['Run'] == run_alias]
+    run = run[run['ReadNumber'] == 1]
+    run = run[~run['SampleID'].isna()]
+    run = run.drop_duplicates(['SampleID', 'LaneNumber'])
```

Comment:
Calculating the `run` DataFrame is done twice (the other time is on Line 184). Move the calculation into `update_layout` and pass the DataFrame into this function.